python/ray/serve/drivers.py

```diff
@@ -146,13 +148,25 @@ def __init__(self, port: int = DEFAULT_GRPC_PORT):
 
     async def run(self):
         """Start gRPC Server"""
-
         logger.info(
             "Starting gRPC server with on node:{} "
-            "listening on port {}".format(ray.util.get_node_ip_address(), self.port)
+            "listening on port {}: {}".format(
+                ray.util.get_node_ip_address(), self.port
+            )
         )
+        address = "[::]:{}".format(self.port)
+        try:
+            self.grpc_port = add_port_to_grpc_server(self.server, address)
```

Comment:
In high level, we should make secured/insecure configurable.

Reply:
I used the add_port_to_grpc_server(self.server, address) function, which internally depends on RAY_USE_TLS environment variable. Does that make it configurable? I can add a comment in my code to indicate this

Reply:
Hi @ashahab , RAY_USE_TLS will trigger all communication internally with TLS, is this the case you are expecting? Or you only want to ingress port to be TLS secured?

Reply:
I suggest having separate variable to control it

Reply:
@sihanwang41 Thank you for your review.
If RAY_USE_TLS is separate from another variable (e.g. RAY_USE_TLS_INGRESS), that may confuse the meaning of "RAY_USE_TLS" and allow insecure communication when it's on.
IMHO it is better to error on the side of caution and ensure all endpoints(ingress, headnode, and worker) encrypt on the wire when RAY_USE_TLS is on. This also allows reusing the `add_port_to_grpc_server` function which already has built in support. Happy to hear more on your thoughts.

Reply:
hi @ashahab , if you set RAY_USE_TLS, all internal communication will be under TLS (potentially hurting performance),  I think you only want to have ingress port as TLS right?
If yes above,
RAY_SERVE_GRPC_TLS_INGRESS = True or RAY_USE_TLS = True, we both set the secure port. Otherwise we use insecure port, what do you think?

Reply:
@sihanwang41 Thank you for the prompt response.

Yes, I agree with you on the potential performance hit with RAY_USE_TLS.
My intent and proposal is to encrypt all the endpoints, and not just the ingress port. From a security perspective, this is the safer approach, given that if the data that is consumed by the endpoint needs to be encrypted between client and RayServe, it is unlikely that it does not need to be encrypted when it's routed from head to worker.

As for performance, I plan to follow this up with some benchmarks and potential improvements on the TLS communication:
1. Ensuring TLS 1.3 is used at each endpoint, reducing handshake time by merging server hello, client key verify, cert, and verify.
2. Session resumption
3. Dynamic record sizes.

Reply:
@sihanwang41 I'd like. your feedback on this. Thank you!